cinema/views.py

```diff
@@ -56,3 +73,31 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+        if date:
+            queryset = queryset.filter(show_time__date=date)
+        if movie:
+            queryset = queryset.filter(movie_id=int(movie))
+        return queryset.distinct()
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.prefetch_related("teckets").select_related(
+        "movie_session"
+    )
```

Comment:
There is a typo in the `OrderViewSet` where `teckets` should be `tickets`. This typo could lead to errors when trying to prefetch related ticket objects.